tests/components/mqtt/test_discovery.py

```diff
@@ -245,6 +251,66 @@
     assert state.state == STATE_ON
 
 
+ABBREVIATIONS_WHITE_LIST = [
+    # MQTT client/server settings
+    'CONF_BIRTH_MESSAGE',
+    'CONF_BROKER',
+    'CONF_CERTIFICATE',
+    'CONF_CLIENT_CERT',
+    'CONF_CLIENT_ID',
+    'CONF_CLIENT_KEY',
+    'CONF_DISCOVERY',
+    'CONF_DISCOVERY_PREFIX',
+    'CONF_EMBEDDED',
+    'CONF_KEEPALIVE',
+    'CONF_TLS_INSECURE',
+    'CONF_TLS_VERSION',
+    'CONF_WILL_MESSAGE',
+    # Undocumented device configuration
+    'CONF_DEPRECATED_VIA_HUB',
+    'CONF_VIA_DEVICE',
+    # Already short
+    'CONF_FAN_MODE_LIST',
+    'CONF_HOLD_LIST',
+    'CONF_HS',
+    'CONF_MODE_LIST',
+    'CONF_PRECISION',
+    'CONF_QOS',
+    'CONF_SCHEMA',
+    'CONF_SWING_MODE_LIST',
+    'CONF_TEMP_STEP',
+]
+
+
+def check_abbreviations_file(path, regex, missing):
+    """Check file for missing abbreviations."""
+    with open(path) as file:
+        matches = re.findall(regex, file.read())
+        for match in matches:
+            if (match[1] not in ABBREVIATIONS.values()
+                    and match[1] not in DEVICE_ABBREVIATIONS.values()
+                    and match[0] not in ABBREVIATIONS_WHITE_LIST):
+                missing.append("{}: no abbreviation for {} ({})".format(
+                    path, match[1], match[0]))
+    return missing
+
+
+async def test_missing_discover_abbreviations(hass, mqtt_mock, caplog):
+    """Check MQTT platforms for missing abbreviations."""
+    missing = []
+    mqtt_dir = '../../../homeassistant/components/mqtt'
+    script_dir = os.path.realpath(
+        os.path.dirname(inspect.getfile(inspect.currentframe())))
```

Comment:
Please rewrite all this file handling with `__file__`, `mqtt.__file`, `PathLib` and `glob`.